lib/dependency-manager-adapters/pnpm.js

```diff
@@ -1,70 +1,18 @@
 'use strict';
 
-const fs = require('fs-extra');
-const path = require('path');
-const debug = require('debug')('ember-try:dependency-manager-adapter:pnpm');
-const Backup = require('../utils/backup');
-const semverLt = require('semver/functions/lt');
-const semverGte = require('semver/functions/gte');
+const semver = require('semver');
+const { LOCKFILE } = require('../utils/package-managers');
+const { BaseAdapter } = require('./base');
 
-const PACKAGE_JSON = 'package.json';
-const PNPM_LOCKFILE = 'pnpm-lock.yaml';
+module.exports = class PnpmAdapter extends BaseAdapter {
+  defaultInstallOptions = ['--no-lockfile', '--ignore-scripts'];
+  lockfile = LOCKFILE.pnpm;
+  name = 'pnpm';
+  overridesKey = 'overrides';
```

Comment:
pnpm overrides key is [nested in a `pnpm` key](https://pnpm.io/package_json#pnpmoverrides) and afaict they do not support [top-level overrides](https://github.com/pnpm/pnpm/issues/7421). 

This bug was here before https://github.com/ember-cli/ember-try/issues/970 and I wonder if it isn't the source of the issue where sometimes people see issues with peer dependencies.

Reply:
The pnpm adapter had this, so I thought it would make sense for all adapters. Will remove!

Reply:
Will fix!

Reply:
thank you <3 

I know it is slightly tangential but your nice refactor incidentally makes it a touch harder to fix, so it seemed to fit now

Reply:
Should be fixed now.